File: Domain/Specialty.cs
namespace Domain
{
	public class Specialty
	{
		public Guid Id { get; set; }
		public SpecialtyBase SpecialtyBase { get; set; }
		public int PriceUAH { get; set; }
		public string Description { get; set; }
		public bool BudgetAllowed { get; set; }
		public int StartYear { get; set; } = DateTime.UtcNow.Year;
		public int EndYear { get; set; } = DateTime.UtcNow.Year + 1;
		public Degree Degree { get; set; }
		public Faculty Faculty { get; set; }
		public ICollection<Language> Languages { get; set; } = new List<Language>();
		public ICollection<StudyForm> StudyForms { get; set; } = new List<StudyForm>();
		public ICollection<EduComponent> EduComponents { get; set; } = new List<EduComponent>();
	}
}

File: client-app/src/app/models/specialty.ts
import { Degree } from "./degree";

export interface Specialty {
	id: string;
	facultyId: string;
	universityId: string;
	specialtyBase: SpecialtyBase;
	priceUAH: number;
	description: string;
	budgetAllowed: boolean;
	ectsCredits: number;
	startYear: number;
	endYear: number;
	degree: Degree;
	languages: Language[];
	studyForms: StudyForm[];
}

export interface SpecialtyBase {
	id: string;
	name: string;
	isceds: Isced[];
}

export interface SpecialtyBaseDto {
	id: string;
	name: string;
}

export interface Isced {
	id: string;
	name: string;
}

export interface Language {
	id: string;
	name: string;
}

export interface StudyForm {
	id: number;
	form: string;
}

export class SpecialtyFormValues {
	id: string = "";
	specialtyBase?: SpecialtyBase;
	priceUAH: number = 0;
	description: string = "";
	budgetAllowed: boolean = false;
	ectsCredits: number = 0;
	degree?: Degree;
	languages?: Language[];
	studyForms?: StudyForm[];
}